contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -129,11 +131,14 @@ contract TrueCreditAgency is UpgradeableClaimable {
     }
 
     function poke(ITrueFiPool2 pool) public {
-        for (uint16 i = 1; i <= MAX_CREDIT_SCORE; i++) {
-            CreditScoreBucket storage bucket = buckets[pool][i];
-            if (bucket.borrowersCount == 0) {
+        uint256 bitMap = usedBuckets;
+        for (uint16 i = 0; i <= MAX_CREDIT_SCORE; (i++, bitMap >>= 1)) {
+            if (bitMap & 1 == 0) {
                 continue;
             }
+
+            CreditScoreBucket storage bucket = buckets[pool][i];
+
             uint256 timeNow = block.timestamp;
```

Comment:
Same as the line above (in a separate PR). The block timestamp can probably be taken out of the inner loop.